src/lib/mdns/Advertiser.h

```diff
@@ -51,20 +51,58 @@ class OperationalAdvertisingParameters
         mPort = port;
         return *this;
     }
-    uint64_t GetPort() const { return mPort; }
+    uint16_t GetPort() const { return mPort; }
 
-    OperationalAdvertisingParameters & EnableIpV4(bool enable)
+private:
+    uint64_t mFabricId = 0;
+    uint64_t mNodeId   = 0;
+    uint16_t mPort     = CHIP_PORT;
+};
+
+class CommissionAdvertisingParameters
+{
+public:
+    CommissionAdvertisingParameters & SetDiscriminator(uint16_t discriminator)
     {
-        mEnableIPv4 = enable;
+        mDiscriminator = discriminator;
         return *this;
     }
-    bool IsIPv4Enabled() const { return mEnableIPv4; }
+    uint16_t GetDiscriminator() const { return mDiscriminator; }
+
+    CommissionAdvertisingParameters & SetVendorId(uint16_t vendorId)
+    {
+        mVendorId = vendorId;
+        return *this;
+    }
+    uint16_t GetVendorId() const { return mVendorId; }
+
+    CommissionAdvertisingParameters & SetProductId(uint16_t productId)
+    {
+        mProductId = productId;
+        return *this;
+    }
+    uint16_t GetProductId() const { return mProductId; }
+
+    CommissionAdvertisingParameters & SetIdentifier(uint64_t identifier)
+    {
+        mIdentifier = identifier;
+        return *this;
+    }
+    uint64_t GetIdentifier() const { return mIdentifier; }
+
+    CommissionAdvertisingParameters & SetPort(uint16_t port)
+    {
+        mPort = port;
+        return *this;
+    }
+    uint16_t GetPort() const { return mPort; }
 
 private:
-    uint64_t mFabricId = 0;
-    uint64_t mNodeId   = 0;
-    uint16_t mPort     = CHIP_PORT;
-    bool mEnableIPv4   = true;
+    uint16_t mDiscriminator;
+    uint16_t mVendorId;
+    uint16_t mProductId;
```

Comment:
Are all these parameters required? I think we use Optional somewhere else. We may also want to initialize these to 0 or something similar.